README.md

```diff
@@ -638,35 +712,131 @@ To negate the meaning of `component_vars.is_filled`, simply treat it as boolean
 {% endif %}
 ```
 
-**Accessing slot names with special characters**
+#### Accessing `is_filled` of slot names with special characters
 
 To be able to access a slot name via `component_vars.is_filled`, the slot name needs to be composed of only alphanumeric characters and underscores (e.g. `this__isvalid_123`).
 
 However, you can still define slots with other special characters. In such case, the slot name in `component_vars.is_filled` is modified to replace all invalid characters into `_`.
 
 So a slot named `"my super-slot :)"` will be available as `component_vars.is_filled.my_super_slot___`.
 
-### Setting Up `ComponentDependencyMiddleware`
+### Scoped slots
```

Comment:
Added section on scoped slots and moved `ComponentDependencyMiddleware` further down to its own section on rendering dependencies.